pandas/core/interchange/from_dataframe.py

```diff
@@ -386,22 +379,17 @@ def datetime_column_to_ndarray(col: Column) -> tuple[np.ndarray | pd.Series, Any
     buffers = col.get_buffers()
 
     _, _, format_str, _ = col.dtype
-    dbuf, data_dtype = buffers["data"]
-
-    if data_dtype[0] == DtypeKind.DATETIME:
-        # temporary workaround to keep backwards compatibility due to
-        # https://github.com/pandas-dev/pandas/issues/54781
-        # Consider dtype being `int` to get number of units passed since 1970-01-01
-        data_dtype = (
-            DtypeKind.INT,
-            data_dtype[1],
-            getattr(ArrowCTypes, f"INT{data_dtype[1]}"),
-            Endianness.NATIVE,
-        )
+    dbuf, _ = buffers["data"]
+    # Consider dtype being `uint` to get number of units passed since the 01.01.1970
 
     data = buffer_to_ndarray(
         dbuf,
-        data_dtype,
+        (
+            DtypeKind.INT,
+            col.dtype[1],
```

Comment:
We unpack `col.dtype` on line 381, it'll be slightly more efficient to get the bit width from there!